src/op_handlers/ret.rs

```diff
@@ -31,6 +31,20 @@ fn get_result(
     Ok(TaggedValue::new_pointer(FatPointer::encode(&result)))
 }
 
+fn save_transient_store(vm: &mut VMState, prev_storage: InMemory) {
```

Comment:
Let's make this return a `Result<(), EraVMError>`, also rewrite these unwraps to proper error handling or use `?`,.